tests/ui/ice-3717.rs

```diff
@@ -0,0 +1,13 @@
+use std::collections::HashSet;
+
+fn main() {}
+
+pub fn ice_3717(_: &HashSet<usize>) {
+    let _ = [0u8; 0];
+    // FIXME: This should produce a suggestion, but we need to bail out here, because
+    // `TypeckTables::expr_ty(..)` panics when we bind an `ExprKind::Repeat(..)` before any `new()`
```

Comment:
lol wat? What's going on here? :laughing: 

Are we maybe overwriting our TypeckTables when going into the `0` constant and not resetting it after leaving said constant? I can't think of a way that the `TypeckTables` would get corrupted in rustc.

Reply:
Ah indeed. in `visit_body` we only set the new body, and don't reset it to the previous body afterwards.

Reply:
Ahh, good catch. Fixed it!